.circleci/config.yml

```diff
@@ -433,44 +442,45 @@ workflows:
   tests:
     jobs:
       - build-manylinux:
-          name: build-manylinux1_<< matrix.architecture >>-py3<< matrix.python-minor >>
+          name: build-<< matrix.manylinux-tag >>_<< matrix.manylinux-arch >>-py<< matrix.python-version >>
           matrix:
             parameters:
-              python-minor: [6, 7, 8, 9]
-              architecture: ["x86_64"]
+              python-version: &python-versions [3.6.8, 3.7.9, 3.8.9, 3.9.4]
+              manylinux-tag: &manylinux-tags ["manylinux1", "manylinux2014"]
+              manylinux-arch: &manylinux-archs ["x86_64"]
       - test-linux:
+          name: test-linux-numpy<< matrix.numpy-version >>-py<< matrix.python-version >>
           # We could break each of these out into seperate jobs so that we
           # don't need to wait for the entire above matrix to complete, but
           # this is cleaner
           requires:
             - build-manylinux
           matrix:
             parameters:
-              # We could make the second numpy version unbounded, but for now
-              # let's pin it for reproducability
-              numpy-version: ['1.17.3', '1.18.5', '1.19.5', '1.20.2']
-              image:
-                - 3.9-buster
-                - 3.8-buster
-                - 3.7-stretch
-                - 3.6-jessie
+              # test the lowest numpy version and the highest minor of each
+              # currently deployed
+              numpy-version: [==1.17.3, <1.19, <1.20, <1.21]
+              python-version: *python-versions
             exclude:
-              - numpy-version: '1.20.2'
-                image: 3.6-jessie
-              - numpy-version: '1.18.5'
-                image: 3.9-buster
-              - numpy-version: '1.17.3'
-                image: 3.9-buster
+              - numpy-version: <1.21
+                image: 3.6.8
```

Comment:
instead of `image`, should this be `python-version` instead?

Reply:
it should. Huh. Wonder how that's working.

Reply:
Interesting, it's ignoring the `--only-binary`. I'll take a look to see what's going on.

Reply:
for posterity, needed `--only-binary=:all:` rather than `--only-binary=all`